### DISCUSSION TITLE
Farfield Plot

### DISCUSSION THREAD

**Discussion by Abrinia** [ansys/pyaedt#4917] [Q&A] [answered]
Hello, 
I am trying to plot farfield data in GUI using the following command:
FF_plot= hfss.post.create_report(expressions="dB(GainTotal)", setup_sweep_name=f"{setup_name[0]} : {sweep_name[1]}", domain='Sweep', variations=vars, primary_sweep_variable="Theta", report_category="Far Fields", plot_type='Rectangular Plot', context= ["Context:=","E_Plane","SourceContext:=", "Source:1"], subdesign_id=None, polyline_points=1001, plot_name="Gain Plot")

But, this command doesn't take a given excitation into account.

How can I define in context field -> farfield sphere along with required source in an array to plot Farfield for all variations by changing  different sources?

Thanks.

**Comment by Samuelopez-ansys** [comment, marked as answer]
If you are interested in far field post processing, please check the new Ffd data classes which give a lot of features.
https://aedt.docs.pyansys.com/version/dev/API/Visualization.html#farfield
